tests/components/homekit_controller/test_sensor.py

```diff
@@ -89,9 +89,8 @@ async def test_temperature_sensor_not_added_twice(hass, utcnow):
     )
 
     for state in hass.states.async_all():
-        if state.entity_id.startswith("button"):
-            continue
-        assert state.entity_id == helper.entity_id
+        if state.entity_id.startswith("sensor"):
```

Comment:
For the future, I'd recommend changing the test to not use conditional logic. Maybe it's enough to look up the expected state and assert its state?

Reply:
Refactored this in my latest push. It gets a list of all temperature sensors and then asserts it against it to a known set. Which is what the test is trying to prove - thats theres only one temperature sensor. Feels better.

Reply:
There's still a conditional check which we want to avoid in tests. Each test should only have one branch. But we don't need to address this in this PR.